specs/electra/p2p-interface.md

```diff
@@ -57,3 +60,33 @@ The following convenience variables are re-defined
 The following validations are added:
 * [REJECT] `len(committee_indices) == 1`, where `committee_indices = get_committee_indices(attestation)`.
 * [REJECT] `attestation.data.index == 0`
+
+## The Req/Resp domain
+
+### Messages
+
+#### BeaconBlocksByRange v3
+
+**Protocol ID:** `/eth2/beacon_chain/req/beacon_blocks_by_range/3/`
+
+Request Content:
+```
+(
+  block_root: Root
```

Comment:
Is this intended to be the last block in the segment  (start_slot + count)? Or just a block anywhere in the segment

Reply:
I'm guessing this is intended to be the target block to sync to?
which could be either greater, equal or less than `(start_slot + count)` but greater than the `start_slot`

e.g. target block_root `0xabcde` (at slot 50)
- `start_slot: 0, count: 32`: valid and returns block for slots 0-31
- `start_slot: 32, count: 32`: valid and returns block for slots 32-50
- `start_slot: 64, count: 32`: invalid request

If the above is correct, perhaps this can be renamed to `target_block_root`?

Reply:
I have updated the definition for `block_root` to refer to a block that's at the tip or descendant of the request slot range 06fb7a0ff8c18269f8387642750a54a5eadca1b7

Reply:
@jimmygchen I like the suggestion to use a more descriptive name than `block_root`. However `target` is overloaded with the CasperFFG checkpoint meaning.

Reply:
I'm thinking that maybe we should just start from the end slot and go back to the ancestors rather than starting from the start slot and go down to the descendants. How about changing the entire request content to the following?

```
(
  block_root: Root
  end_slot: Slot
  count: uint64
)
```

The request will be valid only if `block_root` is in the slot `end_slot`.

Reply:
> The request will be valid only if block_root is in the slot end_slot.

I feel it's an unnecessary departure to how by_range requests are done today